spec/linkedlist.js

```diff
@@ -0,0 +1,238 @@
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedlist'
+
+chai.use(chaiChange)
+
+describe('LinkedList', () => {
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('getHeadNode()', () => {
+    it('Returns the first node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertFirst("Lexus")
+      expect(myLinkedList.getHeadNode().data).to.equal("Lexus")
+    })
+
+    it('Returns the first node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      expect(myLinkedList.getHeadNode().data).to.equal("BMW")
+    })
+  })
+
+  context('getTailNode()', () => {
+    it('Returns the last node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertFirst("Lexus")
+      expect(myLinkedList.getTailNode().data).to.equal("Mercedes")
+    })
+
+    it('Returns the last node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      expect(myLinkedList.getTailNode().data).to.equal("BMW")
+    })
+  })
+
+  context('insert()', () => {
+    it('Inserts a node (with the provided data) to an empty linked list, and sets the head to equal the tail ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      expect(myLinkedList.head).to.equal(myLinkedList.tail)
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.head.data).to.equal("BMW")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.tail.data).to.equal("Tesla")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.head.next.data).to.equal("Tesla")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      expect(myLinkedList.size).to.equal(3)
```

Comment:
this still seems like too many "it"s, from what i've seen the SEP's do on a number of projects is just have a lot of function calls and expects observing the state change in one test.